pkg/appgw/requestroutingrules_test.go

```diff
@@ -233,6 +233,84 @@ var _ = Describe("Test routing rules generations", func() {
 		})
 	})
 
+	Context("test waf policy is configured in rule path", func() {
+		configBuilder := newConfigBuilderFixture(nil)
+		secret := tests.NewSecretTestFixture()
+		endpoint := tests.NewEndpointsFixture()
+		service := tests.NewServiceFixture(*tests.NewServicePortsFixture()...)
+		ingress := tests.NewIngressFixture()
+		ingress.Annotations[annotations.FirewallPolicy] = "/sub/waf"
+
+		It("should have ingress with TLS and redirect", func() {
+			Expect(len(ingress.Spec.TLS) != 0).To(BeTrue())
+			Expect(len(ingress.Spec.TLS[0].SecretName) != 0).To(BeTrue())
+			Expect(ingress.Annotations[annotations.SslRedirectKey]).To(Equal("true"))
+		})
+
+		_ = configBuilder.k8sContext.Caches.Secret.Add(secret)
+		_ = configBuilder.k8sContext.Caches.Endpoints.Add(endpoint)
+		_ = configBuilder.k8sContext.Caches.Service.Add(service)
+		_ = configBuilder.k8sContext.Caches.Ingress.Add(ingress)
+
+		cbCtx := &ConfigBuilderContext{
+			IngressList:           []*v1beta1.Ingress{ingress},
+			ServiceList:           []*v1.Service{service},
+			DefaultAddressPoolID:  to.StringPtr("xx"),
+			DefaultHTTPSettingsID: to.StringPtr("yy"),
+		}
+
+		_ = configBuilder.BackendHTTPSettingsCollection(cbCtx)
+		_ = configBuilder.BackendAddressPools(cbCtx)
+		_ = configBuilder.Listeners(cbCtx)
+
+		// !! Action !! -- will mutate pathMap struct
+		pathMap := configBuilder.getPathMaps(cbCtx)
+
+		rule := &ingress.Spec.Rules[0]
+		listenerID := generateListenerID(ingress, rule, n.HTTP, nil, false)
+		It("has waf policy in pathRule", func() {
+			prs := pathMap[listenerID].ApplicationGatewayURLPathMapPropertiesFormat
+			for _, r := range *prs.PathRules {
+				Expect(r.FirewallPolicy.ID).To(Equal(to.StringPtr("/sub/waf")))
+			}
+		})
```

Comment:
also, check that it is not applied to the listener.

Reply:
> also, check that it is not applied to the listener.

It's covered in front_listener_test.go